metricbeat/mb/testing/data/data_test.go

```diff
@@ -211,7 +212,36 @@ func runTest(t *testing.T, file string, module, metricSetName string, config Con
 		t.Fatalf("could not read file: %s", err)
 	}
 
-	assert.Equal(t, string(expected), string(output))
+	expectedMap := []common.MapStr{}
+	if err := json.Unmarshal(expected, &expectedMap); err != nil {
+		t.Fatal(err)
+	}
+
+	for _, fieldToRemove := range config.RemoveFieldsForComparison {
+		for eventIndex := range data {
+			if err := data[eventIndex].Delete(fieldToRemove); err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		for eventIndex := range expectedMap {
+			if err := expectedMap[eventIndex].Delete(fieldToRemove); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	output, err := json.Marshal(&data)
```

Comment:
@ruflin I modified this from `MarshalIndent` to just `Marshal`. I understood that the idea of using indent was to write into the `data.json` file in pretty format but for comparison it wasn't nice when it failed as the console was escaping newlines like here
![image](https://user-images.githubusercontent.com/4249331/55746270-958a1100-5a39-11e9-88a9-a6abe42e0f86.png)

Reply:
No strong preference as long as the files written to disk are indented. 

For the screenshot you posted: It shows the error pretty well I would say.